crates/services/upgradable-executor/src/instance.rs

```diff
@@ -0,0 +1,574 @@
+use fuel_core_executor::{
+    executor::{
+        ExecutionBlockWithSource,
+        ExecutionOptions,
+    },
+    ports::{
+        MaybeCheckedTransaction,
+        RelayerPort,
+        TransactionsSource,
+    },
+};
+use fuel_core_storage::{
+    column::Column,
+    kv_store::{
+        KeyValueInspect,
+        Value,
+    },
+};
+use fuel_core_types::{
+    blockchain::primitives::DaBlockHeight,
+    fuel_tx::Transaction,
+    fuel_vm::checked_transaction::Checked,
+    services::executor::{
+        Error as ExecutorError,
+        Result as ExecutorResult,
+    },
+};
+use fuel_core_wasm_executor::utils::{
+    pack_exists_size_result,
+    unpack_ptr_and_len,
+    ReturnType,
+};
+use std::{
+    collections::HashMap,
+    sync::Arc,
+};
+use wasmtime::{
+    AsContextMut,
+    Caller,
+    Engine,
+    Func,
+    Linker,
+    Memory,
+    Module,
+    Store,
+};
+
+trait CallerHelper {
+    /// Writes the encoded data to the memory at the provided pointer.
+    fn write(&mut self, ptr: u32, encoded: &[u8]) -> anyhow::Result<()>;
+}
```

Comment:
Why this isn't just part of `Caller`?

Reply:
The `Caller` is the type defined inside of the `wasmtime`.